app/src/main/java/com/rootstrap/android/ui/base/BaseViewModel.kt

```diff
@@ -1,18 +1,26 @@
 package com.rootstrap.android.ui.base
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.rootstrap.android.util.NetworkState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 
 /**
  * A [ViewModel] base class
  * implement app general LiveData as Session or User
  * **/
-open class BaseViewModel : ViewModel() {
-    var error: String? = null
+open class BaseViewModel<UiStateType : UiState>(initialUiState: UiStateType) : ViewModel() {
```

Comment:
I have a doubt here. 
Which will be the difference to save `uiState` as a `Flow` and not as a `LiveData`?

Reply:
When using `Flow`, you're creating channels/routes where data is being sent. I will assume that `Flow` will do it, but when going to another screen check that the channel/route is being paused or destroyed.